instrumentation-annotations-incubator/src/main/java/io/opentelemetry/instrumentation/annotations/incubator/Timed.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.annotations.incubator;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This annotation creates a {@link io.opentelemetry.api.metrics.DoubleHistogram Histogram}
+ * instrument observing the duration of invocations of the annotated method or constructor.
+ *
+ * <p>By default, the Histogram instrument will have the following attributes:
+ *
+ * <ul>
+ *   <li><b>code.namespace:</b> The fully qualified name of the class whose method is invoked.
+ *   <li><b>code.function:</b> The name of the annotated method.
+ *   <li><b>error.type:</b> This is only present if an Exception is thrown, and contains the {@link
+ *       Class#getName name} of the Exception class.
+ * </ul>
+ *
+ * <p>Application developers can use this annotation to signal OpenTelemetry auto-instrumentation
+ * that the Histogram instrument should be created.
+ *
+ * <p>If you are a library developer, then probably you should NOT use this annotation, because it
+ * is non-functional without the OpenTelemetry auto-instrumentation agent, or some other annotation
+ * processor.
+ */
+@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Timed {
+
+  /**
+   * Name of the Histogram metric.
+   *
+   * <p>The name should follow the metric naming rule: <a
+   * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-naming-rule">https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-naming-rule</a>
+   */
+  String value();
+
+  /**
+   * Description for the metric.
+   *
+   * <p>Description strings should follow the metric description rules: <a
+   * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-description">https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-description</a>
+   */
+  String description() default "";
+
+  /**
+   * The unit for the metric.
+   *
+   * <p>Default is seconds.
+   */
+  TimeUnit unit() default TimeUnit.SECONDS;
```

Comment:
do you think this option is important? all otel semconv metrics are reported in seconds with no option to change them

Reply:
I think yes. Actually in production, we more often monitor the latency in millisecond. For example, p99 of the latency should be lower than 200ms.

Reply:
Sorry, I understand your idea now. The user can not change the ways to record a histogram data, so it's no need to expose the unit configuration to user. It's a good catch, I will update it.

Reply:
We would need to expose similar options for all semconv then, so I think maybe it's ok not to include the option, at least to start with.

Reply:
Ok, I am convinced. I agree with that we should keep it simple in the first version. Currently the default unit is `second`, I am not sure if we should change it to milliseconds or just keep it in seconds? do you have any suggestion?

Reply:
https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#instrument-units

> When instruments are measuring durations, seconds (i.e. `s`) SHOULD be used.